File: frontend/lib/schemas/pending_request.ts
/**
 * Pending request zod schemas — re-export of v2 definitions (Wave 3 W3-FE6).
 *
 * 設計仕様書 v0.9 §3.5 / §4.4 / API 契約 v0.1 §7 に対応する v2 schemas は
 * `frontend/lib/schemas/v2/pending_request.ts` に集約されている。
 *
 * 本ファイルは:
 *   1. 既存のフラットな `lib/schemas/{patient,staff,...}` 命名規約と整合性を
 *      取るための **re-export** ファイル (W3-FE6 所有)
 *   2. v2 移行期に `lib/schemas/v2/...` → `lib/schemas/...` の移行ポイントを
 *      集約する（実装手順書 §1 0-C 参照）
 *
 * 直接 v2 ディレクトリを import せず、本モジュール経由で参照すること。
 */

export {
  pendingRequestApproveSchema,
  pendingRequestRejectSchema,
  pendingRequestV2BaseSchema,
  pendingRequestV2CreateSchema,
  pendingRequestV2ReadSchema,
  pendingRequestV2UpdateSchema,
  type PendingRequestApprove,
  type PendingRequestReject,
  type PendingRequestV2Base,
  type PendingRequestV2Create,
  type PendingRequestV2Read,
  type PendingRequestV2Update,
} from './v2/pending_request';

export {
  REQUEST_SCOPE_VALUES,
  REQUEST_STATUS_VALUES,
  REQUEST_TYPE_VALUES,
  requestScopeEnum,
  requestStatusEnum,
  requestTypeEnum,
  type RequestScope,
  type RequestStatus,
  type RequestType,
} from './v2/enums';

File: backend/app/schemas/pending_request.py
"""Pending request schemas — W2-BE5 API surface.

設計仕様書 v0.9 §3.5 / §4.4 / API 契約 §7 に対応する API 層スキーマ。
v2 で確定した型 (`app.schemas.v2.pending_request`) をそのまま再エクスポートし、
業務 API (`app/api/v1/pending_requests.py`) はこのモジュールから読み込む。

監査要件 (§3.5.3):
  **即時反映** の場合も `status="approved"` で同時作成し、業務反映と
  同一トランザクションで処理する (冪等性確保) — 詳細は ``PendingRequestApplier`` 参照。
"""

from __future__ import annotations

from typing import TypeVar

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.v2.enums import (
    RequestScope,
    RequestStatus,
    RequestType,
)
from app.schemas.v2.pending_request import (
    PendingRequestApprove,
    PendingRequestReject,
    PendingRequestV2Base,
    PendingRequestV2Create,
    PendingRequestV2Read,
    PendingRequestV2Update,
)

T = TypeVar("T")


class PaginatedPendingRequest[T](BaseModel):
    """`GET /api/v1/pending-requests` のページング応答 (`Paginated[PendingRequestV2Read]`)."""

    model_config = ConfigDict(extra="forbid")

    items: list[T] = Field(default_factory=list)
    total: int = 0
    limit: int
    offset: int


__all__ = [
    "PaginatedPendingRequest",
    "PendingRequestApprove",
    "PendingRequestReject",
    "PendingRequestV2Base",
    "PendingRequestV2Create",
    "PendingRequestV2Read",
    "PendingRequestV2Update",
    "RequestScope",
    "RequestStatus",
    "RequestType",
]
